src/components/Modals/Tasks/CreateTask.jsx

```diff
@@ -0,0 +1,187 @@
+import Button from '../../Button/Button';
+import CraftInput from '../CraftInput';
+
+import {
+  CREATE_TASK_ONCHANGE,
+  CREATE_TASK_VALIDATE_FIELDS,
+  CREATE_TASK_VALIDATE_FORM,
+  reducerFunc,
+} from './Task-helpers';
+import { TASKS, db, USERS, createElemRef } from '../../../utilities/fb-helpers';
+import debounce from '../../../utilities/debounce';
+
+const newTaskRef = createElemRef(TASKS);
+
+export default class CreateTask extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {
+        title: '',
+        description: '',
+        startDate: '',
+        deadLine: '',
+        selectedUsers: [],
+      },
+      validator: {
+        title: false,
+        startDate: false,
+        deadLine: false,
+        selectedUsers: false,
+      },
+      usersList: [],
+      isValid: false,
+      errors: {
+        titleError: '',
+        startDateError: '',
+        deadLineError: '',
+        selectedUsersError: '',
+      },
+    };
+    this.onChange = this.onChange.bind(this);
+    this.liftUpCreateTask = this.liftUpCreateTask.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.validateFields = this.validateFields.bind(this);
+    this.validateForm = this.validateForm.bind(this);
+  }
+
+  componentDidMount() {
+    db.collection(USERS)
+      .get()
+      .then((querySnapshot) => {
+        const usersList = [];
+        querySnapshot.forEach((doc) => {
+          usersList.push(doc.data());
+        });
+
+        return usersList;
+      })
+      .then((usersList) =>
+        this.setState((prevState) => ({
+          ...prevState,
+          data: {
+            ...prevState.data,
+            id: newTaskRef.id,
+          },
+          usersList,
+        })),
+      )
+      .catch((error) => {
+        console.log('Error reading users collection: ', error);
+      });
+  }
+
+  onChange(event) {
+    const { target } = event;
+    const { name, value } = target;
+    const targetType = target.type;
+    this.setState(
+      (prevState) =>
+        reducerFunc(prevState, {
+          type: CREATE_TASK_ONCHANGE,
+          name,
+          value,
+          targetType,
+        }),
+      debounce(() => {
+        this.validateFields(name, value, targetType);
+      }, 1000),
+    );
+  }
+
+  validateFields(fieldName, fieldValue, targetType) {
+    const state = reducerFunc(this.state, {
+      type: CREATE_TASK_VALIDATE_FIELDS,
+      fieldName,
+      fieldValue,
+      targetType,
+    });
+    this.setState(state, this.validateForm);
+  }
+
+  validateForm() {
+    const state = reducerFunc(this.state, {
+      type: CREATE_TASK_VALIDATE_FORM,
+    });
+    this.setState(state);
+  }
+
+  closeModal() {
+    const { closeFunc } = this.props;
+    closeFunc();
+  }
+
+  liftUpCreateTask() {
+    const { actFunc } = this.props;
+    const { data } = this.state;
+    const newTask = { ...data };
+    actFunc(newTaskRef, newTask);
```

Comment:
could you name it more properly? it's not clear from name what this `act` actually means

Reply:
I have default agreement, that all modals can take 2 functions: actFunc (depends on what modal do) and closeFunc (as follows from name - close modal). In this case actFunc means "Life up create user action" to dispatch it to DB from common page "TASKS". I can name it by this way:
`    case TASK_MODAL_EDIT_TASK:
      modal = <EditTask task={item} closeFunc={closeFunc} liftUpCreateUser={actFunc} />;
      break;`
```suggestion
  liftUpCreateTask() {
    const { liftUpCreateUser } = this.props;
    const { data } = this.state;
    const newTask = { ...data };
    liftUpCreateUser(newTaskRef, newTask);
    this.closeModal();
  }
```